plugins/kubernetes/app/assets/javascripts/kubernetes/controllers/kubernetes_roles_ctrl.js

```diff
@@ -1,5 +1,16 @@
-samson.controller('KubernetesRolesCtrl', function($state, $stateParams) {
+samson.controller('KubernetesRolesCtrl', function($scope, $stateParams, kubernetesService, kubernetesRoleFactory) {
+  $scope.project_id = $stateParams.project_id;
 
-//TODO: Load roles asynchronously
+  $scope.roles = [];
 
+  (function loadKubernetesRoles() {
```

Comment:
Again, the closure syntax is a bit ugly and unncecessary here,  it does save a line instead of creating an 'init' function and calling it, but the syntax doesn't make it worth it IMO.